diff --git a/ext/misc/zipfile.c b/ext/misc/zipfile.c
index a35a26f4d..28c700c16 100644
--- a/ext/misc/zipfile.c
+++ b/ext/misc/zipfile.c
@@ -1623,7 +1623,13 @@ static int zipfileUpdate(
 
     if( rc==SQLITE_OK ){
       zPath = (const char*)sqlite3_value_text(apVal[2]);
+#ifdef MAGMA_ENABLE_FIXES
       if( zPath==0 ) zPath = "";
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+      MAGMA_LOG("%MAGMA_BUG%", zPath == 0);
+#endif
+#endif
       nPath = (int)strlen(zPath);
       mTime = zipfileGetTime(apVal[4]);
     }
