diff --git a/src/resolve.c b/src/resolve.c
index e90e8a768..654bf133e 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -182,22 +182,37 @@ static int areDoubleQuotedStringsEnabled(sqlite3 *db, NameContext *pTopNC){
 Bitmask sqlite3ExprColUsed(Expr *pExpr){
   int n;
   Table *pExTab;
+  Bitmask result;
 
   n = pExpr->iColumn;
   pExTab = pExpr->y.pTab;
   assert( pExTab!=0 );
+#ifdef MAGMA_ENABLE_FIXES
   if( (pExTab->tabFlags & TF_HasGenerated)!=0
    && (pExTab->aCol[n].colFlags & COLFLAG_GENERATED)!=0 
   ){
     testcase( pExTab->nCol==BMS-1 );
     testcase( pExTab->nCol==BMS );
-    return pExTab->nCol>=BMS ? ALLBITS : MASKBIT(pExTab->nCol)-1;
+    result = pExTab->nCol>=BMS ? ALLBITS : MASKBIT(pExTab->nCol)-1;
   }else{
+#endif
     testcase( n==BMS-1 );
     testcase( n==BMS );
     if( n>=BMS ) n = BMS-1;
-    return ((Bitmask)1)<<n;
+    result = ((Bitmask)1)<<n;
+#ifdef MAGMA_ENABLE_FIXES
   }
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+  MAGMA_LOG("%MAGMA_BUG%",
+    MAGMA_AND((pExTab->tabFlags & TF_HasGenerated)!=0,
+    MAGMA_AND((pExTab->aCol[n].colFlags & COLFLAG_GENERATED)!=0,
+    MAGMA_AND(pExTab->nCol >= BMS,
+              result < ALLBITS)))
+  );
+#endif
+#endif
+  return result;
 }
 
 /*
