diff --git a/ssl/record/rec_layer_s3.c b/ssl/record/rec_layer_s3.c
index b4675f3c8c..e3a8a63ad4 100644
--- a/ssl/record/rec_layer_s3.c
+++ b/ssl/record/rec_layer_s3.c
@@ -1469,6 +1469,10 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
 
             memcpy(buf, &(rr->data[rr->off]), n);
             buf += n;
+
+
+        #ifdef MAGMA_ENABLE_FIXES
+
             if (peek) {
                 /* Mark any zero length record as consumed CVE-2016-6305 */
                 if (SSL3_RECORD_get_length(rr) == 0)
@@ -1482,6 +1486,32 @@ int ssl3_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
                     SSL3_RECORD_set_read(rr);
                 }
             }
+
+        #else
+
+            if(!peek){
+                SSL3_RECORD_sub_length(rr, n);
+                SSL3_RECORD_add_off(rr, n);
+                if (SSL3_RECORD_get_length(rr) == 0) {
+                    s->rlayer.rstate = SSL_ST_READ_HEADER;
+                    SSL3_RECORD_set_off(rr, 0);
+                    SSL3_RECORD_set_read(rr);
+            }
+
+            #ifdef MAGMA_ENABLE_CANARIES
+                MAGMA_LOG("MAE114",MAGMA_AND(MAGMA_AND(peek,SSL3_RECORD_get_length(rr) == 0),!SSL3_RECORD_is_read(rr));
+            #endif
+
+        #endif 
+
+
+
+
+
+
+
+
+
             if (SSL3_RECORD_get_length(rr) == 0
                 || (peek && n == SSL3_RECORD_get_length(rr))) {
                 curr_rec++;
