--- a/crypto/asn1/a_d2i_fp.c
+++ b/crypto/asn1/a_d2i_fp.c
@@ -169,9 +169,12 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
             /* suck in slen bytes of data */
             want = slen;
             if (want > (len - off)) {
-                size_t chunk_max = ASN1_CHUNK_INITIAL_SIZE;
-
                 want -= (len - off);
+#ifdef MAGMA_ENABLE_FIXES
+                size_t chunk_max = ASN1_CHUNK_INITIAL_SIZE;
+#else
+                size_t chunk_max = want;
+#endif
                 if (want > INT_MAX /* BIO_read takes an int length */  ||
                     len + want < len) {
                     ASN1err(ASN1_F_ASN1_D2I_READ_BIO, ASN1_R_TOO_LONG);
@@ -193,6 +196,16 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
                     want -= chunk;
                     while (chunk > 0) {
                         i = BIO_read(in, &(b->data[len]), chunk);
+#ifdef MAGMA_ENABLE_CANARIES
+                        /**
+                         * An EOF is encountered before the requested/allocated
+                         * amount of data was received.
+                         * When the fix above is disabled, the allocated buffer
+                         * is resized initially to the reported length, which
+                         * could result in temporary memory exhaustion.
+                         */
+                        MAGMA_LOG("%MAGMA_BUG%", i <= 0);
+#endif
                         if (i <= 0) {
                             ASN1err(ASN1_F_ASN1_D2I_READ_BIO,
                                     ASN1_R_NOT_ENOUGH_DATA);
