--- a/crypto/x509/x509_req.c
+++ b/crypto/x509/x509_req.c
@@ -44,8 +44,13 @@ X509_REQ *X509_to_X509_REQ(X509 *x, EVP_PKEY *pkey, const EVP_MD *md)
         goto err;
 
     pktmp = X509_get0_pubkey(x);
+#ifdef MAGMA_ENABLE_FIXES
     if (pktmp == NULL)
         goto err;
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%",pktmp == NULL);
+#endif
     i = X509_REQ_set_pubkey(ret, pktmp);
     if (!i)
         goto err;
