--- a/crypto/x509/v3_addr.c
+++ b/crypto/x509/v3_addr.c
@@ -84,12 +84,31 @@ static int length_from_afi(const unsigned afi)
  */
 unsigned int X509v3_addr_get_afi(const IPAddressFamily *f)
 {
+#ifdef MAGMA_ENABLE_FIXES
     if (f == NULL
             || f->addressFamily == NULL
             || f->addressFamily->data == NULL
             || f->addressFamily->length < 2)
         return 0;
     return (f->addressFamily->data[0] << 8) | f->addressFamily->data[1];
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+    /**
+    *This bug will do a 1 byte overread when the length is too short
+    */
+    MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(f != NULL, \
+            MAGMA_AND(f->addressFamily != NULL, \
+            MAGMA_AND(f->addressFamily->data != NULL, \
+            f->addressFamily->length < 2)))
+    );
+
+#endif
+    return ((f != NULL &&
+            f->addressFamily != NULL && f->addressFamily->data != NULL)
+            ? ((f->addressFamily->data[0] << 8) | (f->addressFamily->data[1]))
+            : 0);
+
+#endif
 }
 
 /*
