diff --git a/ext/exif/exif.c b/ext/exif/exif.c
index 57ca8fdaa6..d75b50a4e5 100644
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -4019,7 +4019,14 @@ static int exif_process_IFD_in_TIFF(image_info_type *ImageInfo, size_t dir_offse
 		return FALSE;
 	}
 
+#ifdef MAGMA_ENABLE_FIXES
 	if (ImageInfo->FileSize >= 2 && ImageInfo->FileSize - 2 >= dir_offset) {
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+	MAGMA_LOG("%MAGMA_BUG%", dir_offset > SIZE_MAX - 2);
+#endif
+	if (ImageInfo->FileSize >= dir_offset+2) {
+#endif
 		sn = exif_file_sections_add(ImageInfo, M_PSEUDO, 2, NULL);
 #ifdef EXIF_DEBUG
 		exif_error_docref(NULL EXIFERR_CC, ImageInfo, E_NOTICE, "Read from TIFF: filesize(x%04X), IFD dir(x%04X + x%04X)", ImageInfo->FileSize, dir_offset, 2);
