diff --git a/ext/iconv/iconv.c b/ext/iconv/iconv.c
index 26603c26a4..db10c8e50d 100644
--- a/ext/iconv/iconv.c
+++ b/ext/iconv/iconv.c
@@ -1432,9 +1432,16 @@ static php_iconv_err_t _php_iconv_mime_decode(smart_str *pretval, const char *st
 							 * we can do at this point. */
 							if (*(p1 + 1) == '=') {
 								++p1;
+							#ifdef MAGMA_ENABLE_FIXES
 								if (str_left > 1) {
 									--str_left;
 								}
+							#else
+								#ifdef MAGMA_ENABLE_CANARIES
+									MAGMA_LOG("%MAGMA_BUG%", str_left <= 1);
+								#endif
+								--str_left;
+							#endif
 							}
 
 							err = _php_iconv_appendl(pretval, encoded_word, (size_t)((p1 + 1) - encoded_word), cd_pl);
