diff --git a/ext/exif/exif.c b/ext/exif/exif.c
index 8606bdd926..fef41eb5f5 100644
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -4004,8 +4004,14 @@ static int exif_process_IFD_in_TIFF(image_info_type *ImageInfo, size_t dir_offse
 	if (ImageInfo->ifd_nesting_level > MAX_IFD_NESTING_LEVEL) {
 		return FALSE;
 	}
-
+#ifdef MAGMA_ENABLE_FIXES
 	if (ImageInfo->FileSize >= 2 && ImageInfo->FileSize - 2 >= dir_offset) {
+#else
+	if (ImageInfo->FileSize >= dir_offset+2) {
+		#ifdef MAGMA_ENABLE_CANARIES
+			MAGMA_LOG("MAE004", dir_offset == 0xFFFFFFFF);
+		#endif
+#endif
 		sn = exif_file_sections_add(ImageInfo, M_PSEUDO, 2, NULL);
 #ifdef EXIF_DEBUG
 		exif_error_docref(NULL EXIFERR_CC, ImageInfo, E_NOTICE, "Read from TIFF: filesize(x%04X), IFD dir(x%04X + x%04X)", ImageInfo->FileSize, dir_offset, 2);
