diff --git a/ext/xmlrpc/libxmlrpc/xml_element.c b/ext/xmlrpc/libxmlrpc/xml_element.c
index 1678759351..843174afdd 100644
--- a/ext/xmlrpc/libxmlrpc/xml_element.c
+++ b/ext/xmlrpc/libxmlrpc/xml_element.c
@@ -716,9 +716,15 @@ xml_element* xml_elem_parse_buf(const char* in_buf, int len, XML_ELEM_INPUT_OPTI
          long byte_idx = XML_GetCurrentByteIndex(parser);
 /*         int byte_total = XML_GetCurrentByteCount(parser); */
          const char * error_str = (const char *) XML_ErrorString(err_code);
+    #ifdef MAGMA_ENABLE_FIXES
          if(byte_idx > len) {
              byte_idx = len;
          }
+    #else
+         #ifdef MAGMA_ENABLE_CANARIES
+              MAGMA_LOG("MAE001",byte_idx > len);
+         #endif
+    #endif
          if(byte_idx >= 0) {
              snprintf(buf,
                       sizeof(buf),
