diff --git a/codebase/libtiff4/Makefile.am b/codebase/libtiff4/Makefile.am
index d19e58e..0be254d 100644
--- a/codebase/libtiff4/Makefile.am
+++ b/codebase/libtiff4/Makefile.am
@@ -28,6 +28,9 @@ docdir = $(LIBTIFF_DOCDIR)
 AUTOMAKE_OPTIONS = 1.12 dist-zip foreign
 ACLOCAL_AMFLAGS = -I m4
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 docfiles = \
 	COPYRIGHT \
 	ChangeLog \
diff --git a/codebase/libtiff4/build/Makefile.am b/codebase/libtiff4/build/Makefile.am
index c912a42..9916564 100644
--- a/codebase/libtiff4/build/Makefile.am
+++ b/codebase/libtiff4/build/Makefile.am
@@ -24,6 +24,9 @@
 
 # Process this file with automake to produce Makefile.in.
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 EXTRA_DIST = \
 	CMakeLists.txt \
 	README
diff --git a/codebase/libtiff4/configure.ac b/codebase/libtiff4/configure.ac
index 09f959f..7038421 100644
--- a/codebase/libtiff4/configure.ac
+++ b/codebase/libtiff4/configure.ac
@@ -25,7 +25,7 @@ dnl OF THIS SOFTWARE.
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.64)
-AC_INIT([LibTIFF Software],[4.0.10],[tiff@lists.maptools.org],[tiff])
+AC_INIT([LibTIFF Software],[4.0.10magma],[tiff@lists.maptools.org],[tiff])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(m4)
 AC_LANG(C)
@@ -42,7 +42,7 @@ dnl Don't fill the ALPHA_VERSION field, if not applicable.
 LIBTIFF_MAJOR_VERSION=4
 LIBTIFF_MINOR_VERSION=0
 LIBTIFF_MICRO_VERSION=10
-LIBTIFF_ALPHA_VERSION=
+LIBTIFF_ALPHA_VERSION=magma
 LIBTIFF_VERSION=$LIBTIFF_MAJOR_VERSION.$LIBTIFF_MINOR_VERSION.$LIBTIFF_MICRO_VERSION$LIBTIFF_ALPHA_VERSION
 dnl This will be used with the 'make release' target
 LIBTIFF_RELEASE_DATE=`date +"%Y%m%d"`
@@ -116,6 +116,10 @@ AC_PROG_LN_S
 AC_PATH_PROG(CMAKE, cmake)
 
 
+AC_ARG_VAR(TIFF_COPTS,
+   [additional flags for the C compiler, use this for options that would]
+   [cause configure itself to fail])
+
 # Check if LD supports linker scripts, and define automake conditional
 # HAVE_LD_VERSION_SCRIPT if so.  This functionality is currently
 # constrained to compilers using GNU ld on ELF systems or systems
diff --git a/codebase/libtiff4/libtiff/Makefile.am b/codebase/libtiff4/libtiff/Makefile.am
index 14a250a..30ee72b 100644
--- a/codebase/libtiff4/libtiff/Makefile.am
+++ b/codebase/libtiff4/libtiff/Makefile.am
@@ -27,6 +27,9 @@ LIBPORT = $(top_builddir)/port/libport.la
 LIBTIFF = $(top_builddir)/libtiff/libtiff.la
 libtiffincludedir = $(includedir)
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 EXTRA_DIST = \
 	CMakeLists.txt \
 	Makefile.vc \
diff --git a/codebase/libtiff4/port/Makefile.am b/codebase/libtiff4/port/Makefile.am
index 250479f..69fd9df 100644
--- a/codebase/libtiff4/port/Makefile.am
+++ b/codebase/libtiff4/port/Makefile.am
@@ -23,6 +23,9 @@
 
 # Process this file with automake to produce Makefile.in.
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 EXTRA_DIST = \
 	CMakeLists.txt \
 	Makefile.vc \
diff --git a/codebase/libtiff4/test/Makefile.am b/codebase/libtiff4/test/Makefile.am
index 90c2f3d..d66e833 100644
--- a/codebase/libtiff4/test/Makefile.am
+++ b/codebase/libtiff4/test/Makefile.am
@@ -27,6 +27,9 @@ AUTOMAKE_OPTIONS = 1.12 color-tests parallel-tests foreign
 
 LIBTIFF = $(top_builddir)/libtiff/libtiff.la
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 # Environment parameters to be used during tests
 TESTS_ENVIRONMENT = \
 	MAKE="$(MAKE)" \
diff --git a/codebase/libtiff4/tools/Makefile.am b/codebase/libtiff4/tools/Makefile.am
index 3054bee..400faa8 100644
--- a/codebase/libtiff4/tools/Makefile.am
+++ b/codebase/libtiff4/tools/Makefile.am
@@ -26,6 +26,9 @@
 LIBPORT = $(top_builddir)/port/libport.la
 LIBTIFF = $(top_builddir)/libtiff/libtiff.la
 
+TIFF_COPTS := @TIFF_COPTS@
+AM_CFLAGS := ${TIFF_COPTS}
+
 EXTRA_DIST = \
 	CMakeLists.txt \
 	Makefile.vc
diff --git a/codebase/libtiff4/tools/tiffcp.c b/codebase/libtiff4/tools/tiffcp.c
index 8c81aa4f..94f57a0e 100644
--- a/codebase/libtiff4/tools/tiffcp.c
+++ b/codebase/libtiff4/tools/tiffcp.c
@@ -155,6 +155,7 @@ static TIFF* openSrcImage (char **imageSpec)
 int
 main(int argc, char* argv[])
 {
+	magma_init();
 	uint16 defconfig = (uint16) -1;
 	uint16 deffillorder = 0;
 	uint32 deftilewidth = (uint32) -1;
