diff --git a/codebase/libxml2/Makefile.am b/codebase/libxml2/Makefile.am
index 05d1671..75ea675 100644
--- a/codebase/libxml2/Makefile.am
+++ b/codebase/libxml2/Makefile.am
@@ -8,7 +8,7 @@ DIST_SUBDIRS = include . doc example python xstc
 
 AM_CPPFLAGS = -I$(top_builddir)/include -I$(srcdir)/include
 
-AM_CFLAGS = $(EXTRA_CFLAGS) $(THREAD_CFLAGS) $(Z_CFLAGS) $(LZMA_CFLAGS)
+AM_CFLAGS := $(EXTRA_CFLAGS) $(THREAD_CFLAGS) $(Z_CFLAGS) $(LZMA_CFLAGS) $(XML_COPTS)
 
 check_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
                testThreads testC14N testAutomata testRegexp \
diff --git a/codebase/libxml2/configure.ac b/codebase/libxml2/configure.ac
index 92ba25d..2270b6a 100644
--- a/codebase/libxml2/configure.ac
+++ b/codebase/libxml2/configure.ac
@@ -9,8 +9,8 @@ AC_CANONICAL_HOST
 
 LIBXML_MAJOR_VERSION=2
 LIBXML_MINOR_VERSION=9
 LIBXML_MICRO_VERSION=9
-LIBXML_MICRO_VERSION_SUFFIX=
+LIBXML_MICRO_VERSION_SUFFIX=magma
 LIBXML_VERSION=$LIBXML_MAJOR_VERSION.$LIBXML_MINOR_VERSION.$LIBXML_MICRO_VERSION$LIBXML_MICRO_VERSION_SUFFIX
 LIBXML_VERSION_INFO=`expr $LIBXML_MAJOR_VERSION + $LIBXML_MINOR_VERSION`:$LIBXML_MICRO_VERSION:$LIBXML_MINOR_VERSION
 
@@ -89,6 +89,10 @@ fi
 AC_SUBST(VERSION_SCRIPT_FLAGS)
 AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -n "$VERSION_SCRIPT_FLAGS"])
 
+AC_ARG_VAR(XML_COPTS,
+   [additional flags for the C compiler, use this for options that would]
+   [cause configure itself to fail])
+
 dnl
 dnl We process the AC_ARG_WITH first so that later we can modify
 dnl some of them to try to prevent impossible combinations.  This
@@ -396,7 +400,7 @@ else
         # Try pkg-config first so that static linking works.
         PKG_CHECK_MODULES([Z],[zlib],
             [WITH_ZLIB=1],
-            [:])
+            [WITH_ZLIB=0])
     fi
 
     if test "$WITH_ZLIB" = "0"; then
@@ -413,9 +417,14 @@ else
                     esac]
                 else
                     Z_LIBS="-lz"
-                fi])
+                fi],
+                [WITH_ZLIB=0])
             )
     fi
+    # Found the library via either method?
+    if test "x$WITH_ZLIB" != "x1"; then
+      AC_MSG_ERROR([cannot find zlib])
+    fi
 fi
 
 AC_SUBST(Z_CFLAGS)
@@ -435,7 +444,7 @@ else
         # Try pkg-config first so that static linking works.
         PKG_CHECK_MODULES([LZMA],[liblzma],
             [WITH_LZMA=1],
-            [:])
+            [WITH_LZMA=0])
     fi
 
     # If pkg-config failed, fall back to AC_CHECK_LIB. This
@@ -450,9 +459,14 @@ else
                     LZMA_LIBS="-L${LZMA_DIR}/lib -llzma"
                 else
                     LZMA_LIBS="-llzma"
-                fi])
+                fi],
+                [WITH_LZMA=0])
             )
     fi
+    # Found the library via either method?
+    if test "x$WITH_LZMA" != "x1"; then
+      AC_MSG_ERROR([cannot find liblzma])
+    fi
 fi
 
 AC_SUBST(LZMA_CFLAGS)
diff --git a/codebase/libxml2/xmllint.c b/codebase/libxml2/xmllint.c
index 735d951d..a28bd6bb 100644
--- a/codebase/libxml2/xmllint.c
+++ b/codebase/libxml2/xmllint.c
@@ -3104,5 +3104,6 @@ static void deregisterNode(xmlNodePtr node)
 int
 main(int argc, char **argv) {
+    magma_init();
     int i, acount;
     int files = 0;
     int version = 0;
