diff --git a/ext/exif/exif.c b/ext/exif/exif.c
index 8606bdd926..091e32d058 100644
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -3031,6 +3031,7 @@ static int exif_process_user_comment(image_info_type *ImageInfo, char **pszInfoP
 			/* First try to detect BOM: ZERO WIDTH NOBREAK SPACE (FEFF 16)
 			 * since we have no encoding support for the BOM yet we skip that.
 			 */
+		#ifdef MAGMA_ENABLE_FIXES
 			if (ByteCount >=2 && !memcmp(szValuePtr, "\xFE\xFF", 2)) {
 				decode = "UCS-2BE";
 				szValuePtr = szValuePtr+2;
@@ -3039,6 +3040,24 @@ static int exif_process_user_comment(image_info_type *ImageInfo, char **pszInfoP
 				decode = "UCS-2LE";
 				szValuePtr = szValuePtr+2;
 				ByteCount -= 2;
+		#else
+			if (!memcmp(szValuePtr, "\xFE\xFF", 2)) {
+				#ifdef MAGMA_ENABLE_CANARIES
+					MAGMA_LOG("MAE007",!ByteCount < 2);
+				#endif
+                decode = "UCS-2BE";
+                szValuePtr = szValuePtr+2;
+              	ByteCount -= 2;
+			} else if (!memcmp(szValuePtr, "\xFF\xFE", 2)) {
+				#ifdef MAGMA_ENABLE_CANARIES
+					MAGMA_LOG("MAE007",!ByteCount < 2);
+				#endif
+                decode = "UCS-2LE";
+                szValuePtr = szValuePtr+2;
+                ByteCount -= 2;
+
+
+		#endif
 			} else if (ImageInfo->motorola_intel) {
 				decode = ImageInfo->decode_unicode_be;
 			} else {
