diff --git a/utils/pdfunite.cc b/utils/pdfunite.cc
index a8116e37..8227cb60 100644
--- a/utils/pdfunite.cc
+++ b/utils/pdfunite.cc
@@ -214,15 +214,36 @@ int main (int argc, char *argv[])
     intents = catDict->lookup("OutputIntents");
     afObj = catDict->lookupNF("AcroForm").copy();
     Ref *refPage = docs[0]->getCatalog()->getPageRef(1);
+#ifdef MAGMA_ENABLE_FIXES
     if (!afObj.isNull() && refPage) {
+#else
+    if (!afObj.isNull()) {
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("JCH213", !refPage);
+#endif
+#endif
       docs[0]->markAcroForm(&afObj, yRef, countRef, 0, refPage->num, refPage->num);
     }
     ocObj = catDict->lookupNF("OCProperties").copy();
+#ifdef MAGMA_ENABLE_FIXES
     if (!ocObj.isNull() && ocObj.isDict() && refPage) {
+#else
+    if (!ocObj.isNull() && ocObj.isDict()) {
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("JCH213", !refPage);
+#endif
+#endif
       docs[0]->markPageObjects(ocObj.getDict(), yRef, countRef, 0, refPage->num, refPage->num);
     }
     names = catDict->lookup("Names");
+#ifdef MAGMA_ENABLE_FIXES
     if (!names.isNull() && names.isDict() && refPage) {
+#else
+    if (!names.isNull() && names.isDict()) {
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("JCH213", !refPage);
+#endif
+#endif
       docs[0]->markPageObjects(names.getDict(), yRef, countRef, 0, refPage->num, refPage->num);
     }
     if (intents.isArray() && intents.arrayGetLength() > 0) {
