diff --git a/liolib.c b/liolib.c
index b08397da..49d72eeb 100644
--- a/liolib.c
+++ b/liolib.c
@@ -291,7 +291,13 @@ static int io_popen (lua_State *L) {
   const char *filename = luaL_checkstring(L, 1);
   const char *mode = luaL_optstring(L, 2, "r");
   LStream *p = newprefile(L);
+#ifdef MAGMA_ENABLE_FIXES
   luaL_argcheck(L, l_checkmodep(mode), 2, "invalid mode");
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+  MAGMA_LOG("%MAGMA_BUG%", !l_checkmodep(mode));
+#endif
+#endif
   p->f = l_popen(L, filename, mode);
   p->closef = &io_pclose;
   return (p->f == NULL) ? luaL_fileresult(L, 0, filename) : 1;
