diff --git a/src/alaw.c b/src/alaw.c
index 6e0a794..61b0f42 100644
--- a/src/alaw.c
+++ b/src/alaw.c
@@ -327,9 +327,17 @@ s2alaw_array (const short *ptr, int count, unsigned char *buffer)
 static inline void
 i2alaw_array (const int *ptr, int count, unsigned char *buffer)
 {	while (--count >= 0)
+#ifdef MAGMA_ENABLE_FIXES
 	{	if (ptr [count] == INT_MIN)
 			buffer [count] = alaw_encode [INT_MAX >> (16 + 4)] ;
 		else if (ptr [count] >= 0)
+#else
+	{
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("SND015", ptr[count] == INT_MIN);
+#endif
+		if (ptr [count] >= 0)
+#endif
 			buffer [count] = alaw_encode [ptr [count] >> (16 + 4)] ;
 		else
 			buffer [count] = 0x7F & alaw_encode [- ptr [count] >> (16 + 4)] ;
