diff --git a/libtiff/tif_dirread.c b/libtiff/tif_dirread.c
index 1b300b0..b04612d 100644
--- a/libtiff/tif_dirread.c
+++ b/libtiff/tif_dirread.c
@@ -5773,7 +5773,11 @@ static void allocChoppedUpStripArrays(TIFF* tif, uint32 nstrips,
         if (stripbytes > bytecount)
             stripbytes = bytecount;
         newcounts[i] = stripbytes;
+#ifdef MAGMA_ENABLE_FIXES
         newoffsets[i] = stripbytes ? offset : 0;
+#else
+        newoffsets[i] = offset;
+#endif
         offset += stripbytes;
         bytecount -= stripbytes;
     }
@@ -5848,9 +5852,18 @@ ChopUpSingleUncompressedStrip(TIFF* tif)
 	 */
 	if (rowsperstrip >= td->td_rowsperstrip)
 		return;
+#ifdef MAGMA_ENABLE_FIXES
         nstrips = TIFFhowmany_32(td->td_imagelength, rowsperstrip);
         if( nstrips == 0 )
             return;
-
+#else
+        uint64_t nstrips64 = TIFFhowmany_64(bytecount, stripbytes);
+        if ((nstrips64==0)||(nstrips64>0xFFFFFFFF)) /* something is wonky, do nothing. */
+            return;
+        nstrips = (uint32_t)nstrips64;
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("%MAGMA_BUG%", nstrips > TIFFhowmany_32(td->td_imagelength, rowsperstrip));
+#endif
         /* If we are going to allocate a lot of memory, make sure that the */
         /* file is as big as needed */
