diff --git a/HTMLparser.c b/HTMLparser.c
index d6971b5..69047dc 100644
--- a/HTMLparser.c
+++ b/HTMLparser.c
@@ -2787,15 +2787,19 @@ htmlParseAttValue(htmlParserCtxtPtr ctxt) {
 static xmlChar *
 htmlParseSystemLiteral(htmlParserCtxtPtr ctxt) {
     size_t len = 0, startPosition = 0;
+    const xmlChar *q;
     xmlChar *ret = NULL;
 
     if (CUR == '"') {
         NEXT;
 
+#ifdef MAGMA_ENABLE_FIXES
         if (CUR_PTR < BASE_PTR)
             return(ret);
+#endif
         startPosition = CUR_PTR - BASE_PTR;
 
+	q = CUR_PTR;
 	while ((IS_CHAR_CH(CUR)) && (CUR != '"')) {
 	    NEXT;
 	    len++;
@@ -2804,16 +2808,26 @@ htmlParseSystemLiteral(htmlParserCtxtPtr ctxt) {
 	    htmlParseErr(ctxt, XML_ERR_LITERAL_NOT_FINISHED,
 			 "Unfinished SystemLiteral\n", NULL, NULL);
 	} else {
+#ifdef MAGMA_ENABLE_FIXES
 	    ret = xmlStrndup((BASE_PTR+startPosition), len);
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+	    MAGMA_LOG("AAH036", q != (BASE_PTR+startPosition));
+#endif
+	    ret = xmlStrndup(q, CUR_PTR - q);
+#endif
 	    NEXT;
         }
     } else if (CUR == '\'') {
         NEXT;
 
+#ifdef MAGMA_ENABLE_FIXES
         if (CUR_PTR < BASE_PTR)
             return(ret);
+#endif
         startPosition = CUR_PTR - BASE_PTR;
 
+	q = CUR_PTR;
 	while ((IS_CHAR_CH(CUR)) && (CUR != '\'')) {
 	    NEXT;
 	    len++;
@@ -2822,7 +2836,14 @@ htmlParseSystemLiteral(htmlParserCtxtPtr ctxt) {
 	    htmlParseErr(ctxt, XML_ERR_LITERAL_NOT_FINISHED,
 			 "Unfinished SystemLiteral\n", NULL, NULL);
 	} else {
+#ifdef MAGMA_ENABLE_FIXES
 	    ret = xmlStrndup((BASE_PTR+startPosition), len);
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+	    MAGMA_LOG("AAH036", q != (BASE_PTR+startPosition));
+#endif
+	    ret = xmlStrndup(q, CUR_PTR - q);
+#endif
 	    NEXT;
         }
     } else {
@@ -2847,6 +2868,7 @@ htmlParseSystemLiteral(htmlParserCtxtPtr ctxt) {
 static xmlChar *
 htmlParsePubidLiteral(htmlParserCtxtPtr ctxt) {
     size_t len = 0, startPosition = 0;
+    const xmlChar *q;
     xmlChar *ret = NULL;
     /*
      * Name ::= (Letter | '_') (NameChar)*
@@ -2854,10 +2876,13 @@ htmlParsePubidLiteral(htmlParserCtxtPtr ctxt) {
     if (CUR == '"') {
         NEXT;
 
+#ifdef MAGMA_ENABLE_FIXES
         if (CUR_PTR < BASE_PTR)
             return(ret);
+#endif
         startPosition = CUR_PTR - BASE_PTR;
 
+	q = CUR_PTR;
         while (IS_PUBIDCHAR_CH(CUR)) {
             len++;
             NEXT;
@@ -2867,14 +2892,23 @@ htmlParsePubidLiteral(htmlParserCtxtPtr ctxt) {
 	    htmlParseErr(ctxt, XML_ERR_LITERAL_NOT_FINISHED,
 	                 "Unfinished PubidLiteral\n", NULL, NULL);
 	} else {
+#ifdef MAGMA_ENABLE_FIXES
 	    ret = xmlStrndup((BASE_PTR + startPosition), len);
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+	    MAGMA_LOG("AAH036", q != (BASE_PTR+startPosition));
+#endif
+	    ret = xmlStrndup(q, CUR_PTR - q);
+#endif
 	    NEXT;
 	}
     } else if (CUR == '\'') {
         NEXT;
 
+#ifdef MAGMA_ENABLE_FIXES
         if (CUR_PTR < BASE_PTR)
             return(ret);
+#endif
         startPosition = CUR_PTR - BASE_PTR;
 
         while ((IS_PUBIDCHAR_CH(CUR)) && (CUR != '\'')){
@@ -2882,11 +2916,19 @@ htmlParsePubidLiteral(htmlParserCtxtPtr ctxt) {
             NEXT;
         }
 
+	q = CUR_PTR;
 	if (CUR != '\'') {
 	    htmlParseErr(ctxt, XML_ERR_LITERAL_NOT_FINISHED,
 	                 "Unfinished PubidLiteral\n", NULL, NULL);
 	} else {
+#ifdef MAGMA_ENABLE_FIXES
 	    ret = xmlStrndup((BASE_PTR + startPosition), len);
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+	    MAGMA_LOG("AAH036", q != (BASE_PTR+startPosition));
+#endif
+	    ret = xmlStrndup(q, CUR_PTR - q);
+#endif
 	    NEXT;
 	}
     } else {
