diff --git a/HTMLparser.c b/HTMLparser.c
index d6971b5..15da2ae 100644
--- a/HTMLparser.c
+++ b/HTMLparser.c
@@ -2519,6 +2519,7 @@ htmlParseNameComplex(xmlParserCtxtPtr ctxt) {
 	len += l;
 	NEXTL(l);
 	c = CUR_CHAR(l);
+#ifdef MAGMA_ENABLE_FIXES
 	if (ctxt->input->base != base) {
 	    /*
 	     * We changed encoding from an unknown encoding
@@ -2526,6 +2527,10 @@ htmlParseNameComplex(xmlParserCtxtPtr ctxt) {
 	     */
 	    return(htmlParseNameComplex(ctxt));
 	}
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+	MAGMA_LOG("%MAGMA_BUG%", ctxt->input->base != base);
+#endif
     }
 
     if (ctxt->input->cur - ctxt->input->base < len) {
