diff --git a/parser.c b/parser.c
index 04aa717..579a9a7 100644
--- a/parser.c
+++ b/parser.c
@@ -7914,6 +7914,7 @@ xmlParsePEReference(xmlParserCtxtPtr ctxt)
             xmlChar start[4];
             xmlCharEncoding enc;
 
+#ifdef MAGMA_ENABLE_FIXES
 	    if ((entity->etype == XML_EXTERNAL_PARAMETER_ENTITY) &&
 	        ((ctxt->options & XML_PARSE_NOENT) == 0) &&
 		((ctxt->options & XML_PARSE_DTDVALID) == 0) &&
@@ -7922,7 +7923,17 @@ xmlParsePEReference(xmlParserCtxtPtr ctxt)
 		(ctxt->replaceEntities == 0) &&
 		(ctxt->validate == 0))
 		return;
+#endif
 
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(entity->etype == XML_EXTERNAL_PARAMETER_ENTITY, \
+        MAGMA_AND((ctxt->options & XML_PARSE_NOENT) == 0, \
+        MAGMA_AND((ctxt->options & XML_PARSE_DTDVALID) == 0, \
+        MAGMA_AND((ctxt->options & XML_PARSE_DTDLOAD) == 0, \
+        MAGMA_AND((ctxt->options & XML_PARSE_DTDATTR) == 0, \
+        MAGMA_AND(ctxt->replaceEntities == 0, \
+        ctxt->validate == 0)))))));
+#endif
 	    input = xmlNewEntityInputStream(ctxt, entity);
 	    if (xmlPushInput(ctxt, input) < 0) {
                 xmlFreeInputStream(input);
