diff --git a/src/sd2.c b/src/sd2.c
index dd3dbef..b289a71 100644
--- a/src/sd2.c
+++ b/src/sd2.c
@@ -478,12 +478,20 @@ sd2_parse_rsrc_fork (SF_PRIVATE *psf)
 
 	rsrc.str_index = -1 ;
 	for (k = 0 ; k < rsrc.type_count ; k ++)
+#ifdef MAGMA_ENABLE_FIXES
 	{	if (rsrc.type_offset + k * 8 > rsrc.rsrc_len)
 		{	psf_log_printf (psf, "Bad rsrc marker.\n") ;
 			goto parse_rsrc_fork_cleanup ;
 			} ;
 
 		marker = read_rsrc_marker (&rsrc, rsrc.type_offset + k * 8) ;
+#else
+	{
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("SND003", rsrc.type_offset + k * 8 > rsrc.rsrc_len);
+#endif
+		marker = read_marker (rsrc.rsrc_data, rsrc.type_offset + k * 8) ;
+#endif
 
 		if (marker == STR_MARKER)
 		{	rsrc.str_index = k ;
