diff --git a/codebase/poppler/splash/Splash.cc b/codebase/poppler/splash/Splash.cc
index 654608c..2405541 100644
--- a/codebase/poppler/splash/Splash.cc
+++ b/codebase/poppler/splash/Splash.cc
@@ -5862,6 +5862,7 @@ SplashError Splash::blitTransparent(SplashBitmap *src, int xSrc, int ySrc,
     return splashErrZeroImage;
   }
 
+#ifdef MAGMA_ENABLE_FIXES
   if (src->getWidth() - xSrc < width)
     width = src->getWidth() - xSrc;
 
@@ -5879,6 +5880,15 @@ SplashError Splash::blitTransparent(SplashBitmap *src, int xSrc, int ySrc,
 
   if (height < 0)
     height = 0;
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+  MAGMA_LOG(48, (src->getWidth() - xSrc < width) | \
+                (src->getHeight() - ySrc < height) | \
+                (bitmap->getWidth() - xDest < width) | \
+                (bitmap->getHeight() - yDest < height) \
+  );
+#endif
+#endif
 
   switch (bitmap->mode) {
   case splashModeMono1:
