diff --git a/fofi/FoFiTrueType.cc b/fofi/FoFiTrueType.cc
index 24624352..6a48384b 100644
--- a/fofi/FoFiTrueType.cc
+++ b/fofi/FoFiTrueType.cc
@@ -1227,16 +1227,26 @@ void FoFiTrueType::cvtSfnts(FoFiOutputFunc outputFunc,
 	  dumpString(file + tables[j].offset, tables[j].len,
 		     outputFunc, outputStream);
 	} else if (needVerticalMetrics && i == t42VheaTable) {
-	  if (unlikely(length > (int)sizeof(vheaTab))) {
+#ifdef MAGMA_ENABLE_FIXES
+            if (unlikely(length > (int)sizeof(vheaTab))) {
 	    error(errSyntaxWarning, -1, "length bigger than vheaTab size");
 	    length = sizeof(vheaTab);
 	  }
-	  dumpString(vheaTab, length, outputFunc, outputStream);
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+            MAGMA_LOG("JCH208", length > (int)sizeof(vheaTab));
+#endif
+            dumpString(vheaTab, length, outputFunc, outputStream);
 	} else if (needVerticalMetrics && i == t42VmtxTable) {
-	  if (unlikely(length > vmtxTabLength)) {
+#ifdef MAGMA_ENABLE_FIXES
+            if (unlikely(length > vmtxTabLength)) {
 	    error(errSyntaxWarning, -1, "length bigger than vmtxTab size");
 	    length = vmtxTabLength;
 	  }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+            MAGMA_LOG("JCH208", length > vmtxTabLength);
+#endif
 	  dumpString(vmtxTab, length, outputFunc, outputStream);
 	}
       }
