diff --git a/src/printf.c b/src/printf.c
index fc77f68df..ffc49fc5c 100644
--- a/src/printf.c
+++ b/src/printf.c
@@ -586,10 +586,17 @@ void sqlite3_str_vappendf(
         {
           i64 szBufNeeded;           /* Size of a temporary buffer needed */
           szBufNeeded = MAX(e2,0)+(i64)precision+(i64)width+15;
+#ifdef MAGMA_ENABLE_FIXES
           if( szBufNeeded > etBUFSIZE ){
+#else
+          if( MAX(e2,0)+precision+width > etBUFSIZE - 15 ){
+#endif
             bufpt = zExtra = printfTempBuf(pAccum, szBufNeeded);
             if( bufpt==0 ) return;
           }
+#ifdef MAGMA_ENABLE_CANARIES
+          MAGMA_LOG("JCH233", MAX(e2,0)+(i64)precision+(i64)width+15 > etBUFSIZE);
+#endif
         }
         zOut = bufpt;
         nsd = 16 + flag_altform2*10;
