diff --git a/src/build.c b/src/build.c
index 73e4cea7c..593b31617 100644
--- a/src/build.c
+++ b/src/build.c
@@ -2236,10 +2236,15 @@ void sqlite3EndTable(
   ** table itself.  So mark it read-only.
   */
   if( db->init.busy ){
+#ifdef MAGMA_ENABLE_FIXES
     if( pSelect ){
       sqlite3ErrorMsg(pParse, "");
       return;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("JCH220", pSelect);
+#endif
     p->tnum = db->init.newTnum;
     if( p->tnum==1 ) p->tabFlags |= TF_Readonly;
   }
diff --git a/src/prepare.c b/src/prepare.c
index 228d14876..2c3b19c67 100644
--- a/src/prepare.c
+++ b/src/prepare.c
@@ -38,7 +38,14 @@ static void corruptSchema(
     char *z;
     if( zObj==0 ) zObj = "?";
     z = sqlite3MPrintf(db, "malformed database schema (%s)", zObj);
-    if( zExtra && zExtra[0] ) z = sqlite3MPrintf(db, "%z - %s", z, zExtra);
+    if( zExtra
+#ifdef MAGMA_ENABLE_FIXES
+        && zExtra[0]
+#endif
+        ) z = sqlite3MPrintf(db, "%z - %s", z, zExtra);
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("JCH220", zExtra[0]);
+#endif
     *pData->pzErrMsg = z;
     pData->rc = SQLITE_CORRUPT_BKPT;
   }
